File: java/Sorts/SelectionSort.java
public class SelectionSort<E> {

    public <E extends Comparable<E>> void selectionSort(E[] arr){
        int i, j;
        int min;
        E tmp;

        if(arr.length == 0) return;

        for(i = 0; i < arr.length-1; i++){
            min = i;
            for(j = i+1; j < arr.length; j++){
                if(arr[j].compareTo(arr[min]) < 0) min = j;
            }
            if(arr[min].compareTo(arr[i]) != 0){
                tmp = arr[min];
                arr[min] = arr[i];
                arr[i] = tmp;
            }
        }
        return;
    }
}

File: javascript/Sorting_algorithms/selection_sort.js
function selectionSort(arr)
{
    var minIdx, temp, 
    len = arr.length;
    for(var i = 0; i < len; i++)
    {
        minIdx = i;
        for(var  j = i+1; j<len; j++)
        {
            if(arr[j]<arr[minIdx])
            {
                minIdx = j;
            }
        }
        temp = arr[i];
        arr[i] = arr[minIdx];
        arr[minIdx] = temp;
    }
    return arr;
}